nixos/modules/services/misc/matrix-synapse.nix

```diff
@@ -176,548 +169,497 @@ in {
+              description = ''
+                Maximum number of pixels that will be thumbnailed
+              '';
+            };
+
+            dynamic_thumbnails = mkOption {
+              type = types.bool;
+              default = false;
+              example = true;
+              description = ''
+                Whether to generate new thumbnails on the fly to precisely match
+                the resolution requested by the client. If true then whenever
+                a new resolution is requested by the client the server will
+                generate a new thumbnail. If false the server will pick a thumbnail
+                from a precalculated list.
+              '';
+            };
+
+            turn_uris = mkOption {
+              type = types.listOf types.str;
+              default = [];
+              example = [
+                "turn:turn.example.com:3487?transport=udp"
+                "turn:turn.example.com:3487?transport=tcp"
+                "turns:turn.example.com:5349?transport=udp"
+                "turns:turn.example.com:5349?transport=tcp"
+              ];
+              description = ''
+                The public URIs of the TURN server to give to clients
+              '';
+            };
+            turn_shared_secret = mkOption {
+              type = types.str;
+              default = "";
+              example = literalExpression ''
+                config.services.coturn.static-auth-secret
+              '';
+              description = ''
+                The shared secret used to compute passwords for the TURN server.
+              '';
+            };
+
+            trusted_key_servers = mkOption {
+              type = types.listOf (types.submodule {
+                options = {
+                  server_name = mkOption {
+                    type = types.str;
+                    example = "matrix.org";
+                    description = ''
+                      Hostname of the trusted server.
+                    '';
+                  };
+
+                  verify_keys = mkOption {
+                    type = types.nullOr (types.attrsOf types.str);
+                    default = null;
+                    example = literalExpression ''
+                      {
+                        "ed25519:auto" = "Noi6WqcDj0QmPxCNQqgezwTlBKrfqehY1u2FyWP9uYw";
+                      }
+                    '';
+                    description = ''
+                      Attribute set from key id to base64 encoded public key.
+
+                      If specified synapse will check that the response is signed
+                      by at least one of the given keys.
+                    '';
+                  };
+                };
+              });
+              default = [ {
+                server_name = "matrix.org";
+                verify_keys = {
+                  "ed25519:auto" = "Noi6WqcDj0QmPxCNQqgezwTlBKrfqehY1u2FyWP9uYw";
+                };
+              } ];
```

Comment:
This will throw warnings into the log without the user setting some other weird option.

We could also add nixos.org or something to the list here. Not that it'd matter much considering they're both managed by element

Reply:
This is the message you're talking about, and I think it's might fine to leave this as is. People need to make up their mind who to trust in this world. Adding `nixos.org` wouldn't add much value, given who manages it.

```
synapse_homeserver[928180]: This server is configured to use 'matrix.org' as its trusted key server via the
synapse_homeserver[928180]: 'trusted_key_servers' config option. 'matrix.org' is a good choice for a key
synapse_homeserver[928180]: server since it is long-lived, stable and trusted. However, some admins may
synapse_homeserver[928180]: wish to use another server for this purpose.
synapse_homeserver[928180]: To suppress this warning and continue using 'matrix.org', admins should set
synapse_homeserver[928180]: 'suppress_key_server_warning' to 'true' in homeserver.yaml.
synapse_homeserver[928180]: --------------------------------------------------------------------------------
````

Also this message appeared in the previous setup as well. If someone wants to fix this up they should have a good rationale about where to set our default trust anchors.